extensions/rich_text_components/Link/static/oppialink/plugin.js

```diff
@@ -0,0 +1,53 @@
+CKEDITOR.plugins.add('oppialink', {
+  requires: 'widget',
+  init: function(editor) {
+    CKEDITOR.dialog.add('oppialink', this.path + 'dialogs/oppialink.js');
+
+    editor.widgets.add('oppialink', {
+      button: 'Add a link',
+      inline: true,
+      template: '<oppia-noninteractive-link ' +
```

Comment:
(You could pass in a dummy RTE components dict, declared as an oppia.constant, to test this.)